### PR TITLE
Refactor of AMQP-1.0 connector

### PR DESCRIPTION
**Pull request by paramite**
Currently used library is deprecated. This patch makes connector use Azure/go-amqp instead.

### CODE REVIEW

File: connector/amqp10/amqp10.go [hunk 1]
```diff
@@ -10,44 +12,44 @@ import (
...
-	receivers        []AMQP10Receiver
-	senders          map[string]electron.Sender
-	logger           *logging.Logger
-	interrupt        chan bool
+	Address             string
+	ClientName          string
+	SendTimeout         int64
+	ListenPrefetch      int64
+	LinkFailureLimit    int64
+	MaxParalelSendLimit int64
```
Comment (elfiesmelfie) [inline connector/amqp10/amqp10.go]: spelling: parallel

File: connector/amqp10/amqp10.go [hunk 2]
```diff
@@ -10,44 +12,44 @@ import (
...
+	amqp "github.com/Azure/go-amqp"
 )
 
 const (
 	defaultSendTimeout      = 2
-	defaultListenPrefetch   = -1
+	defaultListenPrefetch   = 1
 	defaultClientName       = "localhost"
 	defaultLinkFailureLimit = 20
+	defaultMaxParalelSend   = 128
```
Comment (elfiesmelfie) [inline connector/amqp10/amqp10.go]: spelling: parallel

File: connector/amqp10/amqp10.go (hunk 1 above)
Comment (elfiesmelfie) [reply connector/amqp10/amqp10.go]: s/Paralel/Parallel/g

Comment (vyzigold) [reply connector/amqp10/amqp10.go]: It's also in the config (around L156), not strictly only variables in the code.

File: connector/amqp10/amqp10.go [hunk 3]
```diff
@@ -234,174 +257,273 @@ func (conn *AMQP10Connector) Connect() error {
...
+
+		counter := uint64(0)
+		for {
+			lfLock.RLock()
+			failure := linkFail > conn.LinkFailureLimit
+			lfLock.RUnlock()
+
+			if failure {
+				conn.logger.Warn("Too many link failures in row, reconnecting")
+				err := fmt.Errorf("") //conn.Reconnect("out", outchan, &wg)
```
Comment (vyzigold) [inline connector/amqp10/amqp10.go]: I don't think this and the line after it make sense. Maybe some remnant from debugging?

File: connector/amqp10/amqp10.go [hunk 4]
```diff
@@ -449,167 +572,7 @@ func (conn *AMQP10Connector) Disconnect() {
...
 	conn.startReceivers(outchan, &wg)
-
-	ackchan := make(chan electron.Outcome)
-	linkFail := int64(0)
-	lastAck := int64(0)
-	lfLock := sync.RWMutex{}
-
-	// ACK and error verification goroutine
-	wg.Add(1)
-	go func(ackchan chan electron.Outcome) {
```
Comment (vyzigold) [inline connector/amqp10/amqp10.go:463]: I see the sending goroutine below moved to startSenders(). I don't see the code from this goroutine anywhere anymore. Is the ack check not needed anymore?

Comment (paramite) [reply connector/amqp10/amqp10.go:463]: Unfortunately go-amqp does not have the non-blocking variant of Send method as qpid-proton has. Hence with every send a goroutine waiting for a blocking Send is created. Send blocks until the message is ACKed. There is a timeout send on the Send, so the message will either be ACKed or timeouts.

File: connector/amqp10/amqp10.go (hunk 3 above)
Comment (paramite) [reply connector/amqp10/amqp10.go]: Yes indeed. The reconnect logic is missing. Will add it.

**Review by vyzigold — APPROVED** [review]

**Pull request merged**